Fix build system properly, such that hacking the Makefile.in's
is not required anymore.

--- a/configure.ac
+++ b/configure.ac
@@ -77,6 +77,13 @@
 AC_MSG_RESULT($cgicc_demos)
 AM_CONDITIONAL(DEMOS, test "$cgicc_demos" = yes)
 
+dnl enable/disable the docs, if specified
+AC_ARG_ENABLE([doc],
+    AS_HELP_STRING([--disable-doc], [Disable building documentation]))
+AC_MSG_CHECKING([whether to build the cgicc docs])
+AC_MSG_RESULT([$enable_doc])
+AM_CONDITIONAL([DOCS], [test "x$enable_doc" != "xno"])
+
 if test "$cgicc_demos" = yes; then
   AC_CONFIG_FILES(demo/Makefile)
   AC_CHECK_FUNCS([gethostbyaddr])
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,7 +10,11 @@
 
 ACLOCAL_AMFLAGS= 
 
-SUBDIRS = cgicc doc support $(DEMO)
+SUBDIRS = cgicc support $(DEMO)
+
+if DOCS
+SUBDIRS += doc
+endif
 
 CLEANFILES = *~
 
